src/main.c

```diff
@@ -69,17 +69,51 @@ void connect_console_command(char *str) {
         g->typing = 1;
         g->typing_buffer[0] = '\0';
     } else if (g->server_user[0] == '\0') {
-        strncpy(g->server_user, str, 32);
-        snprintf(g->text_message, KONSTRUCTS_TEXT_MESSAGE_SIZE,
-                "Enter a password to login/create account");
-        snprintf(g->text_prompt, KONSTRUCTS_TEXT_MESSAGE_SIZE, "Password");
+        char new_str[strlen(str)+1]; // Strip all whitespaces away
```

Comment:
@Henningstone Sorry for the long delay. I have been having a pretty bad cold during the week and barely manage to get out of bed. `strnlen` is available from `string.h`  on any POSIX.1-2008 compatible system. That might be a bit to new to be widely available, so it might be a bad idea to use it actually.

Basically I wouldn't say this is very important, it's not like this is going to be an important attack vector, it's just that in general, anything working on strings without `n` in the name can lead to troubles. If the string is not terminated with '\0', any such function will continue through memory until it finds any '\0' or produces a segmentation fault. So it was more of an automatic reaction from my side. A bit silly from my side to be honest. Let's stay with the simple strlen version you did first for now, and sorry for all the delay caused!

I will merge this now. Thank you for your contribution!